File: python/formats.py

import splunklib.client as client
import utils
import sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
if len(sys.argv) != 3:
    print "Syntax: formats.py \"search query\" output_mode"
    sys.exit()
    
try: 
    service = client.connect(**opts.kwargs)
    response = #LAB: call oneshot, pass sys.argv[1] and set the output mode to the value of sys.argv[2]
    print response

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))

File: java/formats.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class formats {

    public static void main(String[] args) {

		try {
			if(args.length != 2) {
				System.out.println("syntax: oneshot \"search query\" output_mode");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);
	        Args searchArgs = new Args();
	        //LAB: add output mode to searchArgs and set it to the value of args[1]
	        InputStream response = //LAB: call oneshot and pass it the query in args[0] and searchArgs

	        int chr = 0;
			while (chr != -1) System.out.write(chr = response.read());
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}